### PR TITLE
Harmonize order state handling

### PR DESCRIPTION
**Pull request by da-kami**
Just some minor cleanup

Harmonize how the order state is updated.
Initially we opted for not exposing `Rejected` because we would exit with an error in the order submission dialog; however, it is not guaranteed that the user will see this error. To keep things simple we map the internal `Rejected` state to `Failed` in the user interface, and update the order accordingly. This guarantees that the user will know that the order failed.

### CODE REVIEW

File: mobile/native/src/trade/order/api.rs [hunk 1]
```diff
@@ -102,9 +102,8 @@ impl From<order::OrderState> for OrderState {
             order::OrderState::Initial => unimplemented!(
                 "don't expose orders that were not submitted into the orderbook to the frontend!"
             ),
-            order::OrderState::Rejected => unimplemented!(
-                "don't expose orders that were rejected by the orderbook to the frontend!"
-            ),
+            // TODO: At the moment the UI does not depict Failed, we could change that
```
Comment (luckysori) [inline mobile/native/src/trade/order/api.rs:105]: ❓ Did you mean that it doesn't depict `Rejected`?

File: mobile/native/src/trade/order/handler.rs [hunk 2]
```diff
@@ -27,16 +27,11 @@ pub async fn submit_order(order: Order) -> Result<()> {
...
+        update_order_state(order.id, OrderState::Rejected)?;
         bail!("Could not post order to orderbook");
     }
-    db::update_order_state(order.id, OrderState::Open)?;
 
-    let order = Order {
-        state: OrderState::Open,
-        ..order
-    };
-    ui_update(order);
```
Comment (luckysori) [inline mobile/native/src/trade/order/handler.rs:39]: ❓ Why do we no longer update the UI with the `order` here? I don't think the commit message talks about this change.

Comment (da-kami) [reply mobile/native/src/trade/order/handler.rs:39]: We do! `update_order_state` handles the UI update, that's why I used it here.

Comment (da-kami) [reply mobile/native/src/trade/order/handler.rs:39]: I added another commit that might help :)

Comment (luckysori) [reply mobile/native/src/trade/order/handler.rs:39]: Ah, nice. Thank you. I clearly didn't understand the first version 😛

**Review by luckysori — APPROVED** [review]

**Review by klochowicz — APPROVED** [review]
👍

**Pull request merged**